src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -103,17 +105,17 @@ public void configure(Map<String, ?> configs) {
         storageCheckInterval = config.getStorageCheckInterval();
         logDirs = config.getLogDirs();
 
-        storageChecker.start();
         log.info("Configured quota callback with {}. Storage quota (soft, hard): ({}, {}). Storage check interval: {}", quotaMap, storageQuotaSoft, storageQuotaHard, storageCheckInterval);
     }
 
     private class StorageChecker implements Runnable {
         private final Thread storageCheckerThread = new Thread(this, "storage-quota-checker");
         private volatile boolean running = false;
 
-        void start() {
+        void startIfNecessary() {
             if (!running) {
                 running = true;
```

Comment:
We can't guarantee that `updateClusterMetadata()` is not called simultaneously from two threads. Since the read/store here is not atomic it's possible that two threads try to start the thread. I guess nothing disastrous will happen (just a stacktrace), but let's switch to using `AtomicBoolean`.

Reply:
Just to clarify @tombentley are you saying we should change the running variable to an AtomicBoolean?

Reply:
Yes.